onedrivesdk/src/main/java/com/onedrive/sdk/http/DefaultHttpProvider.java

```diff
@@ -278,6 +299,16 @@ public void run() {
                     return result;
                 }
 
+                if (isUploadSession(resultClass)) {
+                    mLogger.logDebug("Handling upload session completed");
+                    in = new BufferedInputStream(connection.getInputStream());
+                    if (connection.getResponseCode() == httpCreatedResponseCode) {
+                        Item item = this.handleJsonResponse(in, Item.class);
```

Comment:
It feels inappropriate to me to do json parsing of specific (esp. generated) types at this level... What if chunked upload support is added for a new entity type other than `Item`?